src/olympia/files/tests/test_tasks.py

```diff
@@ -76,6 +78,68 @@ def test_repacking_xpi_files(self):
         # in /tmp.
         assert tmp_dir_before_repacking == sorted(os.listdir(tempfile.gettempdir()))
 
+    @override_switch('enable-manifest-normalization', active=False)
+    def test_does_not_normalize_manifest_json_when_switch_is_inactive(self):
+        upload = self.get_upload('webextension.xpi')
+        fake_results = {'errors': 0}
+        with zipfile.ZipFile(upload.path, 'w') as z:
+            manifest_with_comments = """
+            {
+                // Required
+                "manifest_version": 2,
+                "name": "My Extension",
+                "version": "versionString",
+                // Recommended
+                "description": "haupt\u005fstra\u00dfe"
+            }
+            """
+            z.writestr('manifest.json', manifest_with_comments)
+
+        repack_fileupload(fake_results, upload.pk)
+        upload.reload()
+
+        with zipfile.ZipFile(upload.path) as z:
+            with z.open('manifest.json') as manifest:
+                assert manifest.read().decode() == manifest_with_comments
+
+    @override_switch('enable-manifest-normalization', active=True)
+    def test_normalize_manifest_json(self):
+        upload = self.get_upload('webextension.xpi')
+        fake_results = {'errors': 0}
+        with zipfile.ZipFile(upload.path, 'w') as z:
+            manifest_with_comments = """
+            {
+                // Required
+                "manifest_version": 2,
+                "name": "My Extension",
+                "version": "versionString",
+                // Recommended
+                "description": "haupt\u005fstra\u00dfe"
```

Comment:
Same here. In fact, as it is, this test is checking that entering `haupt_straße` normalizes to `haupt_stra\u00dfe` in the JSON. Ideally we'd test instead that entering `haupt\u005fstra\u00dfe` in the JSON normalizes to `haupt_stra\u00dfe`, removing the superfluous encoding for the ascii character. Doubling the backslashes should do that.